connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectService.scala

```diff
@@ -161,6 +169,52 @@ class SparkConnectService(debug: Boolean)
         userId = request.getUserContext.getUserId,
         sessionId = request.getSessionId)
   }
+
+  // customizedMethodDesc returns a customized MethodDescriptor
+  // with updated request and response marshallers.
+  private def customizedMethodDesc(
+    methodDef: MethodDescriptor[MessageLite, MessageLite]
+  ): MethodDescriptor[MessageLite, MessageLite] = {
+    val requestMarshaller =
+      ProtoLiteUtils.marshallerWithRecursionLimit(
+        methodDef
+          .getRequestMarshaller
+          .asInstanceOf[PrototypeMarshaller[MessageLite]]
+          .getMessagePrototype,
+        PROTOBUF_RECURSION_LIMIT
+      )
+    val responseMarshaller =
+      ProtoLiteUtils.marshallerWithRecursionLimit(
+        methodDef
+          .getResponseMarshaller
+          .asInstanceOf[PrototypeMarshaller[MessageLite]]
+          .getMessagePrototype,
+        PROTOBUF_RECURSION_LIMIT
+      )
+    methodDef.toBuilder(requestMarshaller, responseMarshaller).build()
```

Comment:
nit: Would be more self-documenting as
```
methodDef.toBuilder
  .setRequestMarshaller(requestMarshaller)
  .setResponseMarshaller(responseMarshaller)
  .build()
```
instead of the shorthand method